onnxruntime/core/platform/linux/device_discovery.cc

```diff
@@ -120,6 +121,7 @@ Status GetGpuDeviceFromSysfs(const GpuSysfsPathInfo& path_info, OrtHardwareDevic
 
   // metadata
   gpu_device.metadata.Add("card_idx", MakeString(path_info.card_idx));
+  gpu_device.metadata.Add("bus_id", std::filesystem::read_symlink(sysfs_path / "device").filename().string());  // e.g. 0000:65:00.0
```

Comment:
is it possible to infer whether the GPU is discrete from the bus_id? I'm trying to figure out how to determine that - any pointers would be appreciated.

Reply:
good to know, thanks for sharing that observation